shard/src/ext/events_cog.py

```diff
@@ -54,9 +63,61 @@ def __init__(self, bot):
         self.bot = bot
         self.schedule_messages = {}
         self.poll_messages = {}
+        self.tb_react_events = TbReactEvents(self.bot.asyncpg_wrapper)
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_add(self, payload):
+        await self.on_reaction_update(payload, True)
+
+    @commands.Cog.listener()
+    async def on_raw_reaction_remove(self, payload):
+        await self.on_reaction_update(payload, False)
+
+    async def on_reaction_update(self, payload, add):
+        '''
+        Lumps together on_reaction_add and on_reaction_remove into one thing.
+        add: True means its reaction_add, False, means reaction_remove
+        '''
+        channel = await self.bot.fetch_channel(payload.channel_id)
+        message = await channel.fetch_message(payload.message_id)
```

Comment:
this still really annoys me, cause 90% of the time the message is going to be in the cache. I think it might be worth digging into discord.py to try to access it. (`bot.messages` used to work but I think they removed it?)

Reply:
the problem is that the message in the cache might not be up to date with the latest reactions? I don't know much about this.

Reply:
the api call for the reactions is separate from the call for the message. the reactions is unavoidable, I think, but from the message, all that we actually need is the id, so waiting for the fetch seems dumb